app/code/Magento/SalesRule/Model/Validator.php

```diff
@@ -383,7 +383,7 @@ public function initTotals($items, Address $address)
 
                 foreach ($items as $item) {
                     //Skipping child items to avoid double calculations
-                    if ($item->getParentItemId()) {
+                    if ($item->getParentItem()) {
```

Comment:
Hi @aligent-lturner,
What do you think about checking for both options, like this?
```suggestion
                    if ($item->getParentItemId() || $item->getParentItem()) {
```

Could you also cover your changes with some test?

Reply:
Code has been updated to check both cases, and corresponding unit test has been updated to check different combinations.